README.md

```diff
@@ -1,55 +1,85 @@
-# cloudflare-jwt-verifier
-Cloudflare worker that verifies incoming requests have a valid JSON Web Token (JWT) before the request is forwarded on to your infrastructure
+# Approov Quick Start - Cloudflare Worker
+
+Approov Cloudflare worker that verifies incoming requests have a valid [Approov](https://approov.io) JSON Web Token (JWT) before the request is forwarded on to your infrastructure or to a third party service.
+
+Approov is meant to be used when you want to ensure that only your mobile app can talk to your mobile API, and you can learn in more detail how this is done [here](https://approov.io/approov-in-detail.html).
 
-This use case is in conjunction with the [Approov](https://approov.io) service, although it could be modified to verify any JWT.
 
 ## Why?
 
-There are a couple of reasons for wanting to do this:
- 1. Prevent abuse of your unauthenticated APIs (scraping, malicious activity etc)
- 2. DDoS prevention
+There are a lot of [reasons](https://approov.io/approov-in-detail.html) for wanting to do this, and some of them are:
+
+* Allow the API to validate the request is from a genuine and untampered version of your mobile app
+* Prevent abuse of your APIs (scraping, malicious activity, etc)
+* No need for an API key in the app
+* [Approov Dynamic Certificate Pinning](https://approov.io/docs/latest/approov-usage-documentation/#approov-dynamic-pinning) with all the benefits from traditional implementations, but without their drawbacks
+* DDoS prevention
+
+
+## How it works?
+
+A brief overview on how the Approov cloud service and the Approov Cloudflare worker fit together from a backend perspective.
+
+### Approov Cloud Service
+
+The Approov cloud service attests that a device is running a legitimate version of your mobile application and hasn't been tampered with:
+
+1. If the application looks ok, the Approov cloud service returns a token to your mobile application which is then sent over with any request, and if is validated by the Approov Cloudflare worker the request is processed
```

Comment:
Could be reworded to be simpler - signature validity represents request validity

Reply:
You mean replace the entire mention with for the first point with:

    1. signature validity represents request validity

Reply:
I just mean words to that effect. The current version seems unnecessarily indirect. But its not a big issue.